src/opm/parser/eclipse/EclipseState/Schedule/WellProductionProperties.cpp

```diff
@@ -43,50 +44,64 @@ namespace Opm {
 
     WellProductionProperties WellProductionProperties::history(const WellProductionProperties& prev_properties,
                                                                const DeckRecord& record,
-                                                               const WellProducer::ControlModeEnum controlModeWHISTCL)
+                                                               const WellProducer::ControlModeEnum controlModeWHISTCL,
+                                                               const bool switching_from_injector)
     {
         WellProductionProperties p(record);
         p.predictionMode = false;
 
         // update LiquidRate
         p.LiquidRate = p.WaterRate + p.OilRate;
 
-        const auto& cmodeItem = record.getItem("CMODE");
-        if ( !cmodeItem.defaultApplied(0) ) {
-            namespace wp = WellProducer;
-            auto cmode = wp::ControlModeFromString( cmodeItem.getTrimmedString( 0 ) );
+        if ( record.getItem( "BHP" ).hasValue(0) )
+            p.BHPH = record.getItem("BHP").getSIDouble(0);
+        if ( record.getItem( "THP" ).hasValue(0) )
+            p.THPH = record.getItem("THP").getSIDouble(0);
 
-            // when there is an effective control mode specified by WHISTCL, we always use this control mode
-            if (effectiveHistoryProductionControl(controlModeWHISTCL) ) {
-                cmode = controlModeWHISTCL;
-            }
+        const auto& cmodeItem = record.getItem("CMODE");
+        if ( cmodeItem.defaultApplied(0) ) {
+            const std::string msg = "control mode can not be defaulted for keyword WCONHIST";
+            throw std::invalid_argument(msg);
+        }
 
-            if (effectiveHistoryProductionControl(cmode) ) {
-                p.addProductionControl( cmode );
-                p.controlMode = cmode;
-            } else {
-                const std::string cmode_string = cmodeItem.getTrimmedString( 0 );
-                const std::string msg = "unsupported control mode " + cmode_string + " for WCONHIST";
-                throw std::invalid_argument(msg);
-            }
+        namespace wp = WellProducer;
```

Comment:
That `namespace wp = WellProducer;` line does not have very much value?

Reply:
nope. while it is an old code. It is okay to me either way, as a result, I did not change it.